### PR TITLE
ArdeleanuStefan_Task2

### PR DESCRIPTION
**Pull request by ArdeleanuStefan**

### CODE REVIEW

File: ArdeleanuStefan_Task2/src/DynamicStringArray.cpp [hunk 1]
```diff
@@ -0,0 +1,111 @@
+#include <DynamicStringArray.hpp>
+
+DynamicStringArray::DynamicStringArray()
+{
+    dynamicArray = nullptr;
```
Comment (maxlig79) [inline ArdeleanuStefan_Task2/src/DynamicStringArray.cpp]: Please use initializer list (in general better performance)

File: ArdeleanuStefan_Task2/src/DynamicStringArray.cpp [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+size_t DynamicStringArray::get_size() const
+{
+    return size;
+}
+
+void DynamicStringArray::addEntry(const std::string &input)
+{
+
+    if (size == 0)
+    {
```
Comment (maxlig79) [inline ArdeleanuStefan_Task2/src/DynamicStringArray.cpp]: I don't think this is needed, if you remove the if statement should work.

File: ArdeleanuStefan_Task2/src/DynamicStringArray.cpp [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    for (size_t i = 0; i < size; i++)
+    {
+        if (dynamicArray[i] == input)
+        {
+            ok = true;
+        }
+    }
+
+    if (!ok)
+        return ok;
```
Comment (maxlig79) [inline ArdeleanuStefan_Task2/src/DynamicStringArray.cpp]: Multiple return should be avoided

File: ArdeleanuStefan_Task2/src/DynamicStringArray.cpp [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        dynamicArray[i] = obj.dynamicArray[i];
+}
+
+DynamicStringArray &DynamicStringArray::operator=(const DynamicStringArray &obj)
+{
+    if (this == &obj)
+    {
+        return *this;
+    }
+    dynamicArray = new std::string[obj.get_size()];
```
Comment (maxlig79) [inline ArdeleanuStefan_Task2/src/DynamicStringArray.cpp]: Look like a memory leak

File: ArdeleanuStefan_Task2/src/DynamicStringArray.cpp [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+    size = obj.size;
+    for (size_t i = 0; i < size; ++i)
+        dynamicArray[i] = obj.dynamicArray[i];
+    return *this;
```
Comment (maxlig79) [inline ArdeleanuStefan_Task2/src/DynamicStringArray.cpp]: Please avoid multiple return

File: ArdeleanuStefan_Task2/src/DynamicStringArray.cpp [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    }
+    else
+    {
+        std::string *temp = new std::string[size + 1];
+        for (size_t i = 0; i < size; ++i)
+            temp[i] = dynamicArray[i];
+        temp[size] = input;
+        ++size;
+        delete[] dynamicArray;
+        dynamicArray = nullptr;
```
Comment (maxlig79) [inline ArdeleanuStefan_Task2/src/DynamicStringArray.cpp]: I don't think this is needed

File: ArdeleanuStefan_Task2/src/DynamicStringArray.cpp [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+        {
+            if (dynamicArray[i] == input)
+            {
+                continue;
+            }
+            temp[index] = dynamicArray[i];
+            index++;
+        }
+        delete[] dynamicArray;
+        dynamicArray = nullptr;
```
Comment (maxlig79) [inline ArdeleanuStefan_Task2/src/DynamicStringArray.cpp]: Not needed

File: ArdeleanuStefan_Task2/src/Test.cpp [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    BOOST_CHECK(obj1.getEntry(-1).empty() == 1);
+}
+
+BOOST_AUTO_TEST_CASE(test_3_max)
+{
+    DynamicStringArray obj1;
+    obj1.addEntry("unu");
+    obj1.addEntry("doi");
+    obj1.addEntry("trei");
+    obj1.deleteEntry("doi");
```
Comment (maxlig79) [inline ArdeleanuStefan_Task2/src/Test.cpp]: You should check also the return value of the deleteEntry

**Review by maxlig79 — CHANGES_REQUESTED** [review]

**Review by maxlig79 — APPROVED** [review]

**Review by vlcristi91 — APPROVED** [review]

**Pull request merged**